docs/proposals/makefile-improvement-proposal.md

```diff
@@ -0,0 +1,67 @@
+# Proposal for Makefile improvements
+
+## Introduction
+
+The motivation for this proposal is to enable developers working on Argo Workflows to use build tools in a more reproducible way.
+Currently the Makefile is unfortunately too opinionated and as a result is often a blocker when first setting up Argo Workflows locally.
```

Comment:
Sorry, can you clarify what you mean by "opinionated"?

Reply:
I've found that the Makefile makes assumptions about the system it's running on, for example that the system is either running macOS or Ubuntu. 

It will attempt to use apt to install packages for example. 

I think a build system should just not make any assumptions without proving them to be true. Nix is a hermetic build system/package manger which ensures all assumptions made can be proven.

Reply:
yes, that makes a lot of sense. I guess macos developers probably weren't thinking about users of other OS's.